test/spmd/test_dtensor_redistribute.py

```diff
@@ -0,0 +1,270 @@
+import sys
+import unittest
+import torch
+from torch.distributed.tensor.placement_types import Shard, Replicate
+import torch_xla.runtime as xr
+import torch_xla.distributed.spmd as xs
+import torch_xla
+import numpy as np
+import test_xla_sharding_base
+
+
+class DTensorRedistributeTest(test_xla_sharding_base.XlaShardingTest):
+
+  @classmethod
+  def setUpClass(cls):
+    super().setUpClass()
+    xr.use_spmd()
+
+  def _verify_sharding_spec(self, tensor, expected_devices=None):
+    """Verify tensor sharding spec after mark_step"""
+    torch_xla.core.xla_model.mark_step()
+    sharding_spec = torch_xla._XLAC._get_xla_sharding_spec(tensor)
+    if expected_devices:
+      self.assertIn(expected_devices, sharding_spec)
+    return sharding_spec
+
+  # Test tensor shapes: 0D, 1D, 2D, 3D
+  @unittest.skipIf(xr.global_runtime_device_count() < 2, "Need ≥2 devices")
+  def test_tensor_shapes(self):
+    device_count = xr.global_runtime_device_count()
+    mesh = xs.Mesh(np.arange(device_count), (device_count,))
+
+    shapes_and_specs = [
```

Comment:
can probably use `@pametrized` decorator on the test instead of looping manually